### ISSUE TITLE
Improve text-field searches for messages

### ISSUE THREAD

**Issue by jsorva** [apluslms/mooc-jutut#42]
This issue describes a few somewhat related improvements. It could be split into multiple issues.

Here are some way of improving teacher–student communications by enhancing Jutut’s text-field-based search:

* Enable searching for text written by teachers in responses, not just text written by students. (I expect the same field could be used for both.)

* Enable regex searches (in both Form Content and Exercise Identifier).

* Make the Form Content field look at textual content only, not other content such as MCQ responses, time estimates, etc.

**Comment by etanttila** [comment]
A few follow-up questions:

1. Would it be best to have (1) a separate field of searching student text content and responses written by teachers, (2) having one text search field, which would search from both student and teacher responses, (3) having one text search field, which would have an option of searching just student text content OR both student and teacher responses.
I suspect option (1) would be the easiest to implement from the technical point of view, but not significantly. If that is the desired option, would we want both fields to be visible by default of have the teacher text search field be hidden in the additional filtering options?

2. How should the text search field work? Do we want it to (1) use regex searches all the time, (2) search using regex only if the regex is indicated in the search field somehow (e.g. by surrounding the content with "/"-characters), or (3) include an option next to the search field where the user indicates whether they wish to use a "normal" search or regex?
Also, how should the "normal" search work? If there are several words in the search field separated by spaces, should we search for content, which includes all of them in any order (with possibly other words between and in addition)? Do we also want to support searching with typical search operators, including "AND", "OR", "NOT", parentheses, quotation marks (for exact strings), and an asterisk (to help find variations of a root word).

**Comment by jsorva** [comment]
Based on my experiences, the regex search now works for modules (exercise identifier) but not for form contents. (The symptom is that nothing is found — there’s no error message.)

**Comment by etanttila** [comment]
That is correct, the regex search was implemented for exercise identifiers but not for form contents during the summer. We discussed that for the form content, it would be difficult to have a search which supports both regex searches and normal searches (where word order doesn't matter, etc.), and I understood that if needed to decide between the two, it was preferred to have normal searches (with boolean operators). Thus there is no support for regex searches for the form contents.

That was one of the reasons why I didn't mark this issue as solved in any commits or the PR #59 (in them, I indicated that they partially fix this issue). Markku afterwards closed this issue.
If regex search for the form content search is wanted, I would recommend making a new separate issue (as was done for #68).
